corehq/apps/export/models/deid_export.py

```diff
@@ -0,0 +1,42 @@
+import hashlib
+import uuid
+
+from django.db import models
+
+from corehq.const import ONE_DAY
+from corehq.util.quickcache import quickcache
+
+
+class DeIdMapping(models.Model):
+    domain = models.TextField(max_length=255)
+    hashed_value = models.TextField(max_length=32)
+    deid = models.UUIDField(default=uuid.uuid4)
+
+    class Meta:
+        indexes = [
+            models.Index(fields=['domain', 'hashed_value']),
+        ]
+
+    @classmethod
+    def get_deid(cls, value, doc, domain=None):
+        if doc is not None:
+            # use domain from the couch doc if one was passed in
+            domain = doc['domain']
+
+        return cls._get_deid(value, domain)
+
+    @classmethod
+    @quickcache(['value', 'domain'], timeout=90 * ONE_DAY)
+    def _get_deid(cls, value, domain):
+        hashed_value = cls._hash_value(value)
+        deid_mapping, __ = cls.objects.get_or_create(domain=domain, hashed_value=hashed_value)
+        return str(deid_mapping.deid)
+
+    @staticmethod
+    @quickcache(['value'], timeout=90 * ONE_DAY)
+    def _hash_value(value):
+        if value is None:
+            # None is a de-identifiable value but needs a string to encode for lookup
+            value = ''
```

Comment:
I don't think there's a meaningful different between an empty string and `None` in data this would be de-identifying.

Previously you would get the same output for the both `None` and the string `'None'` and this seems better to me, but if there's any reason to retain the previous behavior I can just call `str(value)` instead.